pkg/services/searchV2/index.go

```diff
@@ -130,12 +130,27 @@ func (i *dashboardIndex) run(ctx context.Context, orgIDs []int64) error {
 	for {
 		select {
 		case doneCh := <-i.syncCh:
+			// Executed on search read requests to make sure index is consistent.
 			lastEventID = i.applyIndexUpdates(ctx, lastEventID)
 			close(doneCh)
 		case <-partialUpdateTimer.C:
+			// Periodically apply updates collected in entity events table.
 			lastEventID = i.applyIndexUpdates(ctx, lastEventID)
 			partialUpdateTimer.Reset(partialUpdateInterval)
+		case <-reIndexSignalCh:
```

Comment:
isn't this branch functionally equivalent to `case <-fullReIndexTimer.C:`? as in, would this work
```
case <-reIndexSignalCh:
     log.info(...)
     fallthrough
case <-fullReIndexTimer.C:
     ....
```

Reply:
You are right - almost identical, ... Though `fallthrough` is not available in `select`.

Maybe refactor it like this:

```go
		case <-reIndexSignalCh:
			// External systems may trigger re-indexing, at this moment provisioning does this.
			i.logger.Info("Full re-indexing due to external signal")
			fullReIndexTimer.Reset(0)
```

Reply:
@ArturWierzbicki refactored a bit in ae8b70bc730802b19f143fd491e4bab1e177e1df

Reply:
looks good!